R/disability.R

```diff
@@ -0,0 +1,181 @@
+#' This table must contain `genc_id` and `er_diagnosis_code` (as ICD-10-CA
+#' alphanumeric code) in long format.
+#' Typically, ER diagnoses should be included when deriving disability in order
+#' to increase sensitivity. However, in certain scenarios, users may choose to
+#' only include IP diagnoses by specifying `erdiag = NULL`. This may be useful
+#' when comparing cohorts with different rates of ER admissions.
+#'
+#' @param component_wise (`logical`)
+#' If `component_wise == FALSE` (default), only a single (global) disability
+#' flag will be returned indicating whether each `genc_id` was diagnosed with
+#' *any* disability (`FALSE/TRUE/NA`).
+#'
+#' If `component_wise == TRUE`, for each `genc_id` with a disability,
+#' all identified disability diagnosis codes are returned in long format. An
+#' additional column `disability_category` is returned containing
+#' one of the following 7 disability categories for each diagnosis:
+#' - Physical disability - Congenital Anomalies
+#' - Physical disability - Musculoskeletal disorders
+#' - Physical disability - Neurological disorders
+#' - Physical disability - Permanent Injuries
+#' - Sensory disabilities - Hearing impairments
+#' - Sensory disabilities - Vision impairments
+#' - Developmental Disabilities
+#'
+#' @section Warning:
+#' This function does not differentiate between diagnosis types. That is, the
+#' disability flags are derived based on all diagnosis codes that are provided
+#' as input to this function. By default, users should include all diagnosis
+#' types to identify disabilities. However, if users wish to include only
+#' certain diagnosis types (e.g., type-M for most responsible discharge
+#' diagnosis), the `ipdiag` and `erdiag` input tables should be filtered
+#' accordingly based on `diagnosis_type` *prior to running this function* (for
+#' more details, see [CIHI diagnosis type definitions](https://www.cihi.ca/sites/default/files/document/diagnosis-type-definitions-en.pdf).
+#'
+#' @return `data.table`
+#' If `component_wise == FALSE`, returns a table with all encounters identified
+#' by the `cohort` table input and an additional derived field `disability`
+#' (`logical`) indicating whether any diagnosis code for a disability was
+#' identified. If a `genc_id` does not have any entry in the diagnosis table at
+#' all, `disability = NA`. Note that this is very rare: If no additional
+#' filtering was performed, >99.9% of `genc_ids` should have an entry in the
+#' `ipdiagnosis` (and >99.9% of `genc_ids` that were admitted via ER should have
+#' an entry in the `erdiagnosis` table).
+#'
+#' If `component_wise == TRUE`, will only return `genc_ids` with a disability.
+#' The output is returned in long format, where each row corresponds to a
+#' disability diagnosis (`diagnosis_code`) and its corresponding
+#' `disability_category` (`character`).
+#'
```

Comment:
The `@return` description is somewhat repetitive of what's described in `@param component_wise`.
I would shorten the description in `@param component_wise` by moving contents more about what is returned to the `return` section. It would be cleaner to have a more concise description for parameters.